charts/tfy-karpenter-config/values.yaml

```diff
@@ -157,7 +157,108 @@ karpenter:
     extraTags: {}
     ## @skip karpenter.defaultNodeTemplate.userData
     # Set this to "" to disable this injection
-    userData: ""
+    userData: |
+      #!/bin/bash
+      set -ex
+
+      version_lte() {
+          printf '%s\n' "$1" "$2" | sort -C -V
+      }
+
+      version_lt() {
+          ! version_lte "$2" "$1"
+      }
+
+      CONTAINERD_VERSION_OUTPUT="$(containerd --version)"
+      IFS=" " read -a  CONTAINERD_VERSION_STRING <<< "${CONTAINERD_VERSION_OUTPUT}"
+      CONTAINERD_VERSION=${CONTAINERD_VERSION_STRING[2]}
+      echo containerd version "${CONTAINERD_VERSION}"
+
+      setup_soci() {( set -ex
+        yum install fuse -y
+        modprobe fuse
+
+        curl \
+        --silent \
+        --show-error \
+        --retry 10 \
+        --retry-delay 1 \
+        -L https://github.com/awslabs/soci-snapshotter/releases/download/v0.4.0/soci-snapshotter-0.4.0-linux-amd64.tar.gz \
+        -o soci-snapshotter-0.4.0-linux-amd64.tar.gz
+
+        tar -C /usr/local/bin -xvf soci-snapshotter-0.4.0-linux-amd64.tar.gz soci soci-snapshotter-grpc
+        rm soci-snapshotter-0.4.0-linux-amd64.tar.gz
+
+        cat > /etc/systemd/system/soci-snapshotter.service << EOF
+      [Unit]
+      Description=soci snapshotter containerd plugin
+      Documentation=https://github.com/awslabs/soci-snapshotter
+      After=network.target
+      Before=containerd.service
+
+      [Service]
+      Type=notify
+      ExecStart=/usr/local/bin/soci-snapshotter-grpc
+      Restart=always
+      RestartSec=5
+
+      [Install]
+      WantedBy=multi-user.target
+      EOF
+
+        systemctl daemon-reload
+        systemctl enable --now soci-snapshotter
+        systemctl status soci-snapshotter
```

Comment:
I think when bootstrap script starts containerd it should also start soci snapshotter right? if we configure that containerd needs soci snapshotter as a dependency?

Reply:
I do not want to patch containerd's service definition.

Reply:
I do not know. Maybe it will work. The goal here is to see whether the soci-snapshooter is up without any issues.
https://www.freedesktop.org/software/systemd/man/latest/systemctl.html#Exit%20status

Reply:
> The goal here is to see whether the soci-snapshooter is up without any issues.

If there are any issues, I will not even patch containerd.